bindings/rust/bench/src/harness.rs

```diff
@@ -217,80 +201,88 @@ pub struct TlsConnPair<C: TlsConnection, S: TlsConnection> {
     server: S,
 }
 
-impl<C: TlsConnection, S: TlsConnection> Default for TlsConnPair<C, S> {
-    fn default() -> Self {
-        Self::new(
-            CryptoConfig::default(),
-            HandshakeType::default(),
-            ConnectedBuffer::default(),
-        )
-        .unwrap()
-    }
-}
-
 impl<C: TlsConnection, S: TlsConnection> TlsConnPair<C, S> {
-    /// Wrap two TlsConnections into a TlsConnPair
-    pub fn wrap(client: C, server: S) -> Self {
-        assert!(
-            client.connected_buffer() == &server.connected_buffer().clone_inverse(),
-            "connected buffers don't match"
-        );
+    pub fn new(client_config: &C::Config, server_config: &S::Config) -> TlsConnPair<C, S> {
```

Comment:
Could this function just take `TlsConnection`? To differentiate from `wrap` maybe you can add a comment mentioning that `wrap` also exists for pre-created connections.

Also seems weird that we create config and then connection in this way since `S` already knows about the `S::Config`.

```
config = S::Config::make_config
conn = S::new_from_config(config)
```

Instead is it possible to just have

```
conn = S::new()
```

Reply:
There is no need to have it take `TlsConnection`, because then it would be exactly the same as `wrap`.

It is true that customers can manually create the connection and then create the pair using wrap, but it ends up being called pretty commonly and adding this method removed that boilerplate.

The problem with the suggested shape is that it ignores the configuration component. 
```
conn = S::new()
```
What should this connection be? What versions of TLS should it support? Should it support session resumption? etc. And even if there is a default that is reasonable for some specific use case, it is unlikely to be a reasonable default for all use cases.